src/workerd/api/sockets.c++

```diff
@@ -206,6 +207,12 @@ jsg::Ref<Socket> connectImplNoOutputLock(
 
   auto& ioContext = IoContext::current();
 
+  // Support calling into arbitrary callbacks for any registered "magic" addresses for which custom
+  // connect() logic is needed.
+  KJ_IF_MAYBE(fn, ioContext.getCurrentLock().getGlobalScope().getConnectOverride(addressStr)) {
```

Comment:
I think these overrides should only apply to the global `connect()`. If `connect()` is called on a specific binding, it should not trigger overrides. Note that `connectImpl()` is shared by both the global connect and specific bindings (`Fetcher` instances), so at present this override will affect bindings.

Reply:
Good point, yeah. Fixed.